application-task-default/src/main/java/com/xwiki/task/internal/job/TaskReferenceRelativizerJob.java

```diff
@@ -0,0 +1,107 @@
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.task.internal.job;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.annotation.InstantiationStrategy;
+import org.xwiki.component.descriptor.ComponentInstantiationStrategy;
+import org.xwiki.job.AbstractJob;
+import org.xwiki.job.Job;
+import org.xwiki.job.event.status.JobStatus;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.query.Query;
+import org.xwiki.query.QueryManager;
+
+import com.xwiki.task.event.TaskRelativizedEvent;
+import com.xwiki.task.event.TaskRelativizingEvent;
+import com.xwiki.task.internal.TaskMacroReferenceMigrator;
+import com.xwiki.task.job.TaskReferenceRelativizerJobRequest;
+import com.xwiki.task.job.TaskReferenceRelativizerJobStatus;
+
+/**
+ * A job that will retrieve all the pages containing task macros and that will change the reference parameters of the
+ * macros into a relative version.
+ *
+ * @version $Id$
+ * @since 3.5.2
+ */
+@Component
+@InstantiationStrategy(ComponentInstantiationStrategy.PER_LOOKUP)
+@Named(TaskReferenceRelativizerJob.JOBTYPE)
+public class TaskReferenceRelativizerJob
+    extends AbstractJob<TaskReferenceRelativizerJobRequest, TaskReferenceRelativizerJobStatus>
+{
+    /**
+     * The identifier of the job.
+     */
+    public static final String JOBTYPE = "taskmanager.relativizetaskreference";
+
+    @Inject
+    private QueryManager queryManager;
+
+    @Inject
+    private DocumentReferenceResolver<String> referenceResolver;
+
+    @Inject
+    private TaskMacroReferenceMigrator referenceMigrator;
+
+    @Override
+    public String getType()
+    {
+        return JOBTYPE;
+    }
+
+    @Override
+    protected TaskReferenceRelativizerJobStatus createNewStatus(TaskReferenceRelativizerJobRequest request)
+    {
+        Job currentJob = this.jobContext.getCurrentJob();
+        JobStatus currentJobStatus = currentJob != null ? currentJob.getStatus() : null;
+        return new TaskReferenceRelativizerJobStatus(request, currentJobStatus, observationManager, loggerManager);
+    }
+
+    @Override
+    protected void runInternal() throws Exception
```

Comment:
Where is this used? I see you do the same inside `TaskReferenceMigratorListener`, without using a job

Reply:
I initially thought of using a job to run the migration. Since the migration is ran at install time (inside a job), there is no need for another job. 

The job might still be used sometime in the future so I thought it would be ok to not delete it.

Reply:
If there are usecases where this could be used in the near future, indeed it can be kept it, but a comment should be added to explain why it isn't used. Otherwise, it could be deleted for now, it isn't that much code outside of what you have in `runInternal`